From e2c50506fee918f95425babbccc7ecf28d3d2f87 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Mon, 9 Feb 2015 16:54:52 +0800
Subject: [PATCH] MLK-10238-2: Revert "drivercore: Bind/unbind power domain on
 probe/remove"

This reverts commit ef2c90dea437f8955b7dc089ff1579c2aa06a6b7.
Signed-off-by: Robin Gong <b38343@freescale.com>
---
 drivers/base/dd.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/drivers/base/dd.c b/drivers/base/dd.c
index 4e812a0..8a8d611 100644
--- a/drivers/base/dd.c
+++ b/drivers/base/dd.c
@@ -23,7 +23,6 @@
 #include <linux/kthread.h>
 #include <linux/wait.h>
 #include <linux/async.h>
-#include <linux/pm_domain.h>
 #include <linux/pm_runtime.h>
 #include <linux/pinctrl/devinfo.h>
 
@@ -288,11 +287,6 @@ static int really_probe(struct device *dev, struct device_driver *drv)
 
 	dev->driver = drv;
 
-	/* If using genpd, bind power domain now before probing */
-	ret = genpd_bind_domain(dev);
-	if (ret)
-		goto probe_failed;
-
 	/* If using pinctrl, bind pins now before probing */
 	ret = pinctrl_bind_pins(dev);
 	if (ret)
@@ -323,7 +317,6 @@ static int really_probe(struct device *dev, struct device_driver *drv)
 probe_failed:
 	devres_release_all(dev);
 	driver_sysfs_remove(dev);
-	genpd_unbind_domain(dev);
 	dev->driver = NULL;
 	dev_set_drvdata(dev, NULL);
 
@@ -537,7 +530,7 @@ static void __device_release_driver(struct device *dev)
 			blocking_notifier_call_chain(&dev->bus->p->bus_notifier,
 						     BUS_NOTIFY_UNBOUND_DRIVER,
 						     dev);
-		genpd_unbind_domain(dev);
+
 	}
 }
 
-- 
1.8.0

